services/src/pro/users/postgres_userdb.rs

```diff
@@ -425,6 +435,17 @@ where
             .await
             .map_err(|_error| error::Error::InvalidSession)?;
 
+        let valid_session: bool = row.get(7);
+
+        let valid_until = if valid_session {
+            row.get(4)
+        } else {
+            self.refresh_tokens(session, &tx)
+                .await
+                .map_err(|_error| Error::InvalidSession)?
```

Comment:
is there a more meaningful error here? When can the token refresh fail? if the refresh token is not valid anymore?

Reply:
ee8a2fc34b0071a07bfcd86d003e9da7adb57ff2

I put a log debug message with the original error. My intention of mapping the error was that session_by_id call either gets a session or returns InvalidSession to whomever calls the function. I believe most API handlers use this as a first step, and they probably shouldn't care or propagate to the user that some internal refresh mechanism failed? But you are right, it should be noted somewhere, so I opted for a log for now.

When can this fail: 
The token refresh can fail if the token is not valid anymore. Or if the client does not respond for any reason. The other call to refresh_tokens is from get_access_token, where there might be no refresh_token (InvalidSession).